pkg/ssh/cli.go

```diff
@@ -121,53 +176,35 @@ func (cli *CLI) AddPortalCommand(ctrl ChannelControlInterface) {
 			for r := range ctrl.AllSSHRoutes() {
 				routes = append(routes, fmt.Sprintf("%s@%s", *ctrl.Username(), strings.TrimPrefix(r.From, "ssh://")))
 			}
-			items := []list.Item{}
-			for _, route := range routes {
-				items = append(items, item(route))
+			env := &sshEnviron{
+				Env: map[string]string{
+					"TERM": cli.ptyInfo.TermEnv,
+				},
 			}
-			l := list.New(items, itemDelegate{}, int(cli.ptyInfo.WidthColumns-2), int(cli.ptyInfo.HeightRows-2))
-			l.Title = "Connect to which server?"
-			l.SetShowStatusBar(false)
-			l.SetFilteringEnabled(false)
-			l.Styles.Title = titleStyle
-			l.Styles.PaginationStyle = paginationStyle
-			l.Styles.HelpStyle = helpStyle
-
-			cli.SendTeaMsg(tea.WindowSizeMsg{Width: int(cli.ptyInfo.WidthColumns), Height: int(cli.ptyInfo.HeightRows)})
-			close(cli.msgQueue)
-
-			cli.tui = tea.NewProgram(model{list: l},
+			prog := tui.NewPortalProgram(cmd.Context(), routes, int(cli.ptyInfo.WidthColumns-2), int(cli.ptyInfo.HeightRows-2),
```

Comment:
if a client would send us width/height as 1/0 we will overflow and allocate huge buffer here ?

Reply:
Yes good catch, there are a few places where we need to clamp when converting from ptyInfo here.